lib/ecto/migration/schema_migration.ex

```diff
@@ -33,16 +34,16 @@ defmodule Ecto.Migration.SchemaMigration do
     |> Map.put(:prefix, prefix)
   end
 
-  def up(repo, version, prefix) do
+  def up(repo, repo_name, version, prefix) do
     %__MODULE__{version: version}
     |> Ecto.put_meta(prefix: prefix, source: get_source(repo))
-    |> repo.insert!(@opts)
+    |> (&(Ecto.Repo.Schema.insert!(repo_name, &1, @opts))).()
```

Comment:
Nitpick: please don't use this pipeline style. :) You can either remove the pipe or add a `repo_insert` function that receives arguments in the proper order.

Reply:
👍